dora/src/vm/annotations.rs

```diff
@@ -131,4 +132,25 @@ impl AnnotationDefinition {
             .name;
         annotation_usages.contains(name)
     }
+
+    pub fn reject_modifiers(
```

Comment:
I think we should define what modifiers we allow?

Reply:
I think we talked about this already. With user-definable annotations, the set of all annotations isn't fixed, so flipping the check would require more awkward logic, not less, because we would have do to additional checks with regard to internal vs. user-defined.

Reply:
Tbh, I don't really care anymore, this isn't code that will be with us for a lot further.

Reply:
I don't see how this would be more awkward, the logic would just be `is_user_defined() || is_allowed_internal_annotation()`. Also we probably don't allow all user-defined annotations in all locations in the future (e.g. an annotion which is only allowed on methods).